components/Dialogs/src/MDCDialogPresentationController.m

```diff
@@ -146,6 +137,20 @@ - (void)presentationTransitionWillBegin {
   // presentation and dismissal of the dialog.
   // https://material.io/guidelines/motion/choreography.html#choreography-creation
 
+  // Ensure corner radius matches between the tracking view and the view being presented
+  if (self.dialogCornerRadius < 0) {
```

Comment:
Maybe using a private BOOL ivar dialogCornerRadiusIsSet would be more clear? Using negative number might be fragile?

Reply:
It would be a private variable, so it shouldn't affect API. And adding it would help the clearance of the code, so the number of variable is not a concern in this case. `-1` implicitly delivered the same idea.

Reply:
dialogCornerRadius is a public API, so when clients modify it, these 2 will get out of sync.  The necessary syncing logic to resolve this doesn't payoff - because you now have to test for this -1 in a different place.
Instead I added a public constant to make the use of this case more clear. Please see updated comment in MDCDialogPresentationController.h - hopefully this issue is explained better now.

Reply:
I understand the syncing logic, what I meant was we could add a private instance variable like `isDialogCornerRadiusManuallySet` to avoid the constant issue, because using a default value as a way to indicate a instance hasn't been changed is not stable.

Reply:
I agree with your concern, but im not sure your suggestions will solve it.  This condition now compares with "MDCUsePresentedViewCornerRadius" instead of -1, which makes it a little clearer for the next person (using a constant to imitate swift's nil value, which unfortunately doesn't exist in obj c floats.).  
MDCUsePresentedViewCornerRadius is used in more than one place, and creating 2 variables to test this condition seems like an overkill.
How strongly are you feeling about this?

Reply:
Followed up offline. I feel not exposing the `-1` value for clients would reduce the complexity and remove the unnecessary functionality we provide.